interval/interval.go

```diff
@@ -15,9 +15,38 @@
 // Package interval provides a struct representing a time interval.
 package interval
 
-import "time"
+import (
+	"math/rand"
+	"time"
+)
 
 // Interval represents the interval [Start, End).
 type Interval struct {
 	Start, End time.Time
 }
+
+// RandomInstant returns a random time (to second precision) that falls within
+// the Interval.
+//
+// If the Interval is nil, or Interval.End <= Interval.Start, the zero value
+// time.Time will be returned.
+func (i *Interval) RandomInstant() time.Time {
+	if i == nil {
+		return time.Time{}
+	}
+
+	if i.Start.Equal(i.End) || i.Start.After(i.End) {
+		return time.Time{}
+	}
+
+	start := i.Start.Unix()
+	end := i.End.Unix()
+	delta := end - start
+
+	if delta == 0 {
```

Comment:
Oh look!  You deal with the `iStart == iEnd` case right here.
How about above, just returning `time{}` for the `iEnd < iStart` edge case.

Reply:
This is a nuance, and maybe it's not quite right/is complicating things unecessarily - if we've got here then Interval.Start != Interval.End, but Interval.Start.Unix() == Interval.End.Unix(), implying there is less than a second's difference between Interval.Start and Interval.End, as the .Unix() is only to second precision.

Reply:
Might have another think about this change, and even if it stays this way I'll add some comments.

Reply:
Yeah, there's something here that's incorrect (have found a test case that fails) - will fix things and let you know when it's better :)

Reply:
Okay I think I've fixed it.  I've also added comments explaining what's going on, and I've renamed the method to RandomSecond as it chooses a specific second.  The reason I need it to do that is because the NotAfter field in certificates (which this is being used for) are specified to second precision.